examples/uart-nanopb/main.cpp

```diff
@@ -0,0 +1,301 @@
+// RIOT includes
+#include "periph/uart.h"
+#include "ringbuffer.h"
+#include "riot/chrono.hpp"
+#include "riot/thread.hpp"
+#include "thread.h"
+#include "xtimer.h"
+
+// Projet includes
+#include "shell_menu/shell_menu.hpp"
+
+#include <pb_encode.h>
+#include <pb_decode.h>
+#include "pingpong.pb.h"
+
+#define READER_PRIO        (THREAD_PRIORITY_MAIN - 1)
+#define SENDER_PRIO        (THREAD_PRIORITY_MAIN - 1)
+
+static kernel_pid_t reader_pid, sender_pid;
+static char reader_stack[THREAD_STACKSIZE_MAIN];
+static char sender_stack[THREAD_STACKSIZE_MAIN];
+uart_t uart_dev = UART_DEV(1);
+uint32_t uart_speed = 230400U;
+bool suspend_sender = false;
+
+typedef struct {
+    char rx_mem[PB_Response_size];
+    ringbuffer_t rx_buf;
+} uart_ctx_t;
+
+static uart_ctx_t ctx;
+
+static uint8_t msg_length_bytes_number = 4;
+static uint8_t msg_length_bytes_received = 0;
+static uint32_t msg_length = 0;
+static uint32_t msg_bytes_received = 0;
```

Comment:
Add some comments here please, to know the difference between all these variables.